### PR TITLE
Update rexml dependency (CVE-2024-39908)

### PR DESCRIPTION
**Pull request by bmedenwald**
There is a DoS vulnerability in REXML gem. This vulnerability has been assigned the CVE identifier [CVE-2024-39908](https://www.cve.org/CVERecord?id=CVE-2024-399086). We strongly recommend upgrading the REXML gem.

**Details**
When it parses an XML that has many specific characters such as <, 0 and %>. REXML gem may take long time.

Please update REXML gem to version 3.3.2 or later.

**Affected versions**
REXML gem 3.3.2 or prior

### CODE REVIEW

**Comment by marknorgren** [comment]
Is there a reason why https://github.com/CocoaPods/Xcodeproj/pull/944 wasn't merged?

This would have avoided users having to wait on this PR to merge to update REXML right?

File: xcodeproj.gemspec [hunk 1]
```diff
@@ -26,7 +26,7 @@ Gem::Specification.new do |s|
   s.add_runtime_dependency 'claide',         '>= 1.0.2', '< 2.0'
   s.add_runtime_dependency 'colored2',       '~> 3.1'
   s.add_runtime_dependency 'nanaimo',        '~> 0.3.0'
-  s.add_runtime_dependency 'rexml',          '~> 3.2.4'
+  s.add_runtime_dependency 'rexml',          '~> 3.3.2'
```
Comment (marknorgren) [inline xcodeproj.gemspec:29]:
```suggestion
  s.add_runtime_dependency 'rexml',          '>= 3.2.4', '< 4.0'
```

Would this be helpful to allow users to upgrade to newer minor and patch releases in the future?

Comment (bmedenwald) [reply xcodeproj.gemspec]: Done. Can this get done quick? Taking a long time for a trivial but big security concern.

Comment (marknorgren) [reply xcodeproj.gemspec]: I am waiting on this merging too, I do not have maintainer rights to merge this.

**Review by setoelkahfi — COMMENTED** [review]
We're also waiting for this, @segiddins are you aware of this?

**Comment by samfranz** [comment]
Yup, we need this too.

**Review by marknorgren — APPROVED** [review]

**Comment by iosdevben** [comment]
Thanks for raising this PR and getting it merged.

When are we likely to see a new release of Xcodeproj that incorporates this change?

**Comment by AliSoftware** [comment]
ICYMI: [`Xcodeproj` `1.25.0` has now been released 3 days ago](https://github.com/CocoaPods/Xcodeproj/releases/tag/1.25.0) 🎉

**Pull request merged**